src/main/java/com/olegpruh/bookstore/controller/BookController.java

```diff
@@ -0,0 +1,61 @@
+package com.olegpruh.bookstore.controller;
+
+import com.olegpruh.bookstore.dto.BookSearchParametersDto;
+import com.olegpruh.bookstore.dto.request.CreateBookRequestDto;
+import com.olegpruh.bookstore.dto.response.BookDto;
+import com.olegpruh.bookstore.service.BookService;
+import java.util.List;
+import javax.validation.Valid;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/books")
+public class BookController {
+    private final BookService bookService;
+
+    @Autowired
+    public BookController(BookService bookService) {
+        this.bookService = bookService;
+    }
+
+    @GetMapping
+    public List<BookDto> getAll() {
+        return bookService.findAll();
+    }
+
+    @GetMapping("/{id}")
+    public BookDto getBookById(@PathVariable Long id) {
+        return bookService.findBookById(id);
+    }
+
+    @PostMapping
+    public BookDto createBook(@RequestBody @Valid CreateBookRequestDto bookDto) {
+        return bookService.save(bookDto);
+    }
+
+    @ResponseStatus(HttpStatus.NO_CONTENT)
+    @DeleteMapping("/{id}")
+    public void deleteBook(@PathVariable Long id) {
+        bookService.deleteById(id);
+    }
+
+    @PutMapping("/{id}")
+    public BookDto updateBook(@PathVariable Long id, @RequestBody CreateBookRequestDto requestDto) {

```

Comment:
requestDto wont be validated because of lack of trigger, e.g. @Valid. 
```suggestion
    public BookDto updateBook(@PathVariable Long id, @Valid @RequestBody CreateBookRequestDto requestDto) {

```